parsers/src/main/scala/uk/gov/ons/addressIndex/crfscala/CrfFeature.scala

```diff
@@ -0,0 +1,105 @@
+package uk.gov.ons.addressIndex.crfscala
+
+import uk.gov.ons.addressIndex.crfscala.CrfFeatureAnalyser.CrfFeatureAnalyser
+import uk.gov.ons.addressIndex.crfscala.CrfScala._
+import scala.util.control.NonFatal
+
+/** todo scaladoc
+  *
+  * @tparam T the return type of the FeatureAnalyser
+  */
+trait CrfFeature[T] {
+
+  /**
+    * @return a function which returns an instance of T
+    */
+  def analyser() : CrfFeatureAnalyser[T]
+
+  /**
+    * @return name
+    */
+  def name() : String
```

Comment:
Same as above, a bit of duplication in traits as well that could be delegated to a source `trait`, e.g something that both these classes extend.